crypto/threads_pthread.c

```diff
@@ -281,10 +282,26 @@ static void free_rcu_thr_data(void *ptr)
     CRYPTO_THREAD_set_local(&rcu_thr_key, NULL);
 }
 
+#ifdef FIPS_MODULE
 static void ossl_rcu_init(void)
 {
-    CRYPTO_THREAD_init_local(&rcu_thr_key, NULL);
+    int ossl_unused rc;
+
+    rc = CRYPTO_THREAD_init_local(&rcu_thr_key, NULL);
+    assert(rc == 1);
+    rc = ossl_init_thread_start(NULL, NULL, free_rcu_thr_data);
```

Comment:
This looks suspect the second argument in this call should be an OSSL_LIB_CTX.....there is no default libctx in the FIPS module.

Reply:
oh, you know what, with the recent context changes in rcu, this now conflicts and needs fixing up.  will do

Reply:
rebased, and fixed